app/src/main/java/app/envelop/ui/faq/FaqActivity.kt

```diff
@@ -0,0 +1,80 @@
+package app.envelop.ui.faq
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.webkit.*
+import androidx.core.view.isVisible
+import app.envelop.R
+import app.envelop.ui.BaseActivity
+import app.envelop.ui.common.MessageManager
+import kotlinx.android.synthetic.main.activity_faq.*
+import kotlinx.android.synthetic.main.shared_appbar.*
+import javax.inject.Inject
+
+class FaqActivity : BaseActivity() {
+
+  @Inject
+  lateinit var messageManager: MessageManager
+  private var hadError = false
+
+  @SuppressLint("SetJavaScriptEnabled")
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    component.inject(this)
+    setContentView(R.layout.activity_faq)
+    toolbar.enableNavigation()
+    faq.settings.javaScriptEnabled = true
+    faq.loadUrl(getString(R.string.faq_url))
+    faq.webChromeClient = CustomChromeClient()
+    faq.webViewClient = CustomWebViewClient()
+  }
+
+  private fun openBrowser(url: Uri?) {
+    url?.let {
+      val intent = Intent(Intent.ACTION_VIEW, url)
+      startActivity(intent)
+    }
+  }
+
+  inner class CustomChromeClient : WebChromeClient() {
+    override fun onProgressChanged(view: WebView?, newProgress: Int) {
+      progressBar.progress = newProgress
+      super.onProgressChanged(view, newProgress)
+    }
+  }
+
+  inner class CustomWebViewClient : WebViewClient() {
+    override fun onReceivedError(
+      view: WebView,
+      request: WebResourceRequest,
+      error: WebResourceError
+    ) {
+      hadError = true
+      messageManager.showError(R.string.faq_host_not_found)
+      super.onReceivedError(view, request, error)
```

Comment:
If `super` needs to be called, it's usually at the start or at the end of a method implementation. In this case, it's probably at the end.